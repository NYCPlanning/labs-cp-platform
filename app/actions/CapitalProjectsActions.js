import dispatcher from '../dispatcher';

const CapitalProjectsActions = {
  onFilterDimensionChange(filterDimension, values) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_FILTERDIMENSION_CHANGE',
      filterDimension,
      values,
    });
  },

  fetchDetailData(id) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_FETCH_DETAIL_DATA',
      id,
    });
  },

  resetFilter() {
    dispatcher.dispatch({
      type: 'CAPTIALPROJECTS_RESET_FILTER',
    });
  },

  setSelectedFeatures(features) {
    dispatcher.dispatch({
      type: 'CAPTIALPROJECTS_SET_SELECTED_FEATURES',
      features,
    });
  },
};

export default CapitalProjectsActions;
