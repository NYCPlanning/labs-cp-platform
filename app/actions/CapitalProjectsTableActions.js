import dispatcher from '../dispatcher';

const CapitalProjectsTableActions = {
  onFilterDimensionChange(filterDimension, values) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_FILTERDIMENSION_CHANGE',
      filterDimension,
      values,
    });
  },

  onFilterBy(filterBy) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_FILTERBY',
      filterBy,
    });
  },

  onSortChange(columnKey, sortDir) {
    dispatcher.dispatch({
      type: 'CAPITALPROJECTS_TABLE_SORTCHANGE',
      columnKey,
      sortDir,
    });
  },
};

export default CapitalProjectsTableActions;
