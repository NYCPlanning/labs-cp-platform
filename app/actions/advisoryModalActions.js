import * as AT from '../constants/actionTypes';

export const hideModal = () => ({
  type: AT.HIDE_MODAL,
});

export const showModal = () => ({
  type: AT.SHOW_MODAL,
});
