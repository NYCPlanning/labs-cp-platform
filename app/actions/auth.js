import * as AT from '../constants/actionTypes';

export const login = ({ params }) => ({
  type: AT.AUTH0_LOGIN,
  payload: { params },
});

export const logout = () => ({
  type: AT.AUTH0_LOGOUT,
});

export const signUp = () => ({
  type: AT.AUTH0_SIGNUP,
});

export const loadCredentials = ({ targetPath }) => ({
  type: AT.LOAD_CREDENTIALS,
  payload: { targetPath },
});

export const authorizeUser = (profile, token) => ({
  type: AT.AUTHORIZE_USER,
  payload: { profile, token },
});

export const deauthorizeUser = () => ({
  type: AT.DEAUTHORIZE_USER,
});

export const ensureAccess = (permission) => ({
  type: AT.ENSURE_ACCESS,
});
