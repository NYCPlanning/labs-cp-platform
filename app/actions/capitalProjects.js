import * as AT from '../constants/actionTypes';
import * as cartoActions from './carto';
import { sqlConfig, unionSql } from '../helpers/sqlbuilder/CapitalProjectsSqlBuilder';

export const fetchDetails = capitalProjectId =>
  cartoActions.getFeature({
    tableName: sqlConfig.combinedTable,
    column: 'maprojid',
    value: capitalProjectId,
  }, AT.FETCH_CAPITAL_PROJECT_DETAILS);

export const fetchBudgets = capitalProjectId => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT * FROM cpdb_budgets WHERE maprojid = '${capitalProjectId}'`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_BUDGETS,
  },
});

export const fetchCommitments = capitalProjectId => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT * FROM cpdb_commitments WHERE maprojid = '${capitalProjectId}' ORDER BY to_date(plancommdate,'MM/YY')`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_COMMITMENTS,
  },
});

export const fetchTotalPointsCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM ${sqlConfig.pointsTablename}`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TOTAL_POINTS_COUNT,
  },
});

export const fetchTotalPolygonsCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM ${sqlConfig.polygonsTablename}`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TOTAL_POLYGONS_COUNT,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT count(*) FROM (${unionSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_SELECTED_COUNT,
  },
});

export const setSelectedFeatures = selectedFeatures => ({
  type: AT.SET_SELECTED_CAPITAL_PROJECTS_FEATURES,
  payload: { selectedFeatures },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_CAPITAL_PROJECTS_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const resetFilters = () => ({
  type: AT.RESET_CAPITAL_PROJECTS_FILTERS,
});
