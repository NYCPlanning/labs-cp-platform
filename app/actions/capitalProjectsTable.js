import * as AT from '../constants/actionTypes';
import { getTableSql } from '../helpers/sqlbuilder/CapitalProjectsSqlBuilder';

export const fetchDetails = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: getTableSql(filterDimensions),
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECT_TABLE_DETAILS,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*) FROM (${getTableSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_CAPITAL_PROJECTS_TABLE_SELECTED_COUNT,
  },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const resetFilter = () => ({
  type: AT.RESET_CAPITAL_PROJECTS_TABLE_FILTER,
});

export const setTableSort = (columnKey, sortDir) => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_SORT,
  payload: { columnKey, sortDir },
});

export const setTableFilterBy = filterBy => ({
  type: AT.SET_CAPITAL_PROJECTS_TABLE_FILTER_BY,
  payload: { filterBy },
});
