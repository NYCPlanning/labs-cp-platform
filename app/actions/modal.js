import * as AT from '../constants/actionTypes';

export const openModal = modalOptions => ({
  type: AT.OPEN_MODAL,
  payload: { modalOptions },
});

export const closeModal = () => ({
  type: AT.CLOSE_MODAL,
});
