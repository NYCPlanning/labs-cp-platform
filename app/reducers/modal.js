import * as AT from '../constants/actionTypes';

const initialState = null;

const modalReducer = (state = initialState, action) => {
  switch (action.type) {
    case AT.OPEN_MODAL:
      return action.payload.modalOptions;

    case AT.CLOSE_MODAL:
      return initialState;

    default:
      return state;
  }
};

export default modalReducer;
