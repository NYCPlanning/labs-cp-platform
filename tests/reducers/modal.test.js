import * as actions from '../../app/actions/modal';
import modalReducer from '../../app/reducers/modal';

describe('reducers/modalReducer', () => {
  test('OPEN_MODAL', () =>
    expect(modalReducer(false, actions.openModal({
      modalHeading: 'Modal Heading',
      modalContent: 'Modal Content',
      modalCloseText: 'Close Text',
    }))).toMatchSnapshot());

  test('OPEN_MODAL - reopen existing modal', () => {
    const initialState = {
      modalHeading: 'Modal Heading',
      modalContent: 'Modal Content',
      modalCloseText: 'Close Text',
    };

    expect(modalReducer(initialState, actions.openModal({
      modalHeading: 'New Modal Heading',
      modalContent: 'New Modal Content',
      modalCloseText: 'New Close Text',
    }))).toMatchSnapshot();
  });

  test('CLOSE_MODAL', () => {
    expect(modalReducer(false, actions.closeModal())).toMatchSnapshot();
    expect(modalReducer(true, actions.closeModal())).toMatchSnapshot();
  });
});
