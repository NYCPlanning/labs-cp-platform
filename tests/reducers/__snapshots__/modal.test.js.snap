// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/modalReducer CLOSE_MODAL 1`] = `null`;

exports[`reducers/modalReducer CLOSE_MODAL 2`] = `null`;

exports[`reducers/modalReducer OPEN_MODAL - reopen existing modal 1`] = `
Object {
  "modalCloseText": "New Close Text",
  "modalContent": "New Modal Content",
  "modalHeading": "New Modal Heading",
}
`;

exports[`reducers/modalReducer OPEN_MODAL 1`] = `
Object {
  "modalCloseText": "Close Text",
  "modalContent": "Modal Content",
  "modalHeading": "Modal Heading",
}
`;
